File: 0897-increasing-order-search-tree/0897-increasing-order-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */

let curr;
function Inorder(root){
    if(root == null) return;
    Inorder(root.left);

    root.left = null;
    curr.right = root;
    curr = root;
    Inorder(root.right);
}
var increasingBST = function(root) {
    let tem = new TreeNode(0);
    curr = tem;
    Inorder(root);
    return tem.right;
};

File: 0897-increasing-order-search-tree/0897-increasing-order-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    TreeNode curr;
    public void InOrder(TreeNode root){
        if(root == null) return;
        InOrder(root.left);

        root.left = null;
        curr.right = root;
        curr = root;

        InOrder(root.right); 
    }
    public TreeNode increasingBST(TreeNode root) {
        TreeNode node = new TreeNode(0);
        curr = node;
        InOrder(root);
        return node.right;
    }
}